lib/Dashboard/Model/AMQP.pm

```diff
@@ -18,7 +18,17 @@ use Mojo::Base -base, -signatures;
 
 has [qw(log jobs)];
 
-sub handle ($self, $key, $data) {
+# called from startup in AMQP watcher to catch up with jobs started/ended while the watcher wasn't alive
+sub fetch_old ($self) {
+  my $db = $self->pg->db;
+  my $id = $db->query("SELECT min(job_id) from openqa_jobs where status='waiting'")->hash->{min};
```

Comment:
That doesn't work, if there's no result you get `undef` back and can't access the hash for `->{min}`.

Reply:
This is brute force anyway and I guess we should be smarter about this. I just don't know how :)

In production DB we have a waiting job with an id 2M away from HEAD - so we would fetch 2M jobs to fix up.